.github/actions/yarn-install/action.yml

```diff
@@ -0,0 +1,27 @@
+name: Install Dependencies
+inputs:
+  node-version-type:
+    required: false
+    default: 'current'
+    type: choice
+    options:
+      - 'current'
+      - 'min-supported'
+  min-supported-node-version:
+    required: false
+    default: '18.12'
+  current-node-version:
+    required: false
+    default: '20.2'
```

Comment:
I reckon we can reduce to `18.x`, `20.x` as specifiers at the GitHub level so we can simplify this. Then we could receive one `node-version` arg as a string, and more easily test 3 versions in future if we need to.

Reply:
I would like to have a test on `18.0` though since we claim to support it, and there are some differences vs later minors. `['current', 'lts/*', 'lts/-1', '18.0']` (which resolves to 21.x, 20.x, 18.x, 18.0), IMO.

Reply:
Sounds good to me, however @huntie disagreed that it makes sense to test the minimum and lts versions before.
WDYT @huntie?

Reply:
BTW @robhogan, `current` resolves to `v22.6.0`.
https://github.com/facebook/metro/actions/runs/10286511772/job/28467353526

Reply:
Yeah that's fair - essentially it's unstable (though fwiw these CI signals aren't land blocking yet). I'd like to test 18.0 (which should be stable), I'd be OK to use `22.x`, `20.x` instead of the aliases.

It *would* still be good to know asap if we're incompatible with a new Node major, because we don't put a "maximum" constraint on `engines` so Node releases can break users (it's [happened before](https://github.com/facebook/metro/pull/752)), but ideally we'd run that as a separate scheduled (eg nightly) run rather than anything that potentially signals a diff-time failure unrelated to the diff. We can follow up with that.

Reply:
OK. switched to `['18.0', '18.x', '20.x']` and added to my TODO list to introduce a nightly job that tests `lts/*` and `current`.

Reply:
One more for the nightly TODO list then - we should have a test that doesn't use (maybe deletes) `yarn.lock` and pulls the latest versions of all our dependencies, so we can see if a dependency update breaks us somehow.

Reply:
np

Reply:
@huntie please approve as well